packages/core/src/RenderingEngine/helpers/createVolumeActor.ts

```diff
@@ -43,11 +48,45 @@ async function createVolumeActor(
   const volumeActor = vtkVolume.newInstance();
   volumeActor.setMapper(volumeMapper);
 
+  const voiRange = volumeActor
+    .getProperty()
+    .getRGBTransferFunction(0)
+    .getRange()
+    .slice();
+
   if (callback) {
     callback({ volumeActor, volumeId });
   }
 
+  triggerVOIModifiedIfNecessary(element, viewportId, volumeActor, voiRange);
+
   return volumeActor;
 }
 
+const triggerVOIModifiedIfNecessary = (
+  element: HTMLDivElement,
+  viewportId: string,
+  volumeActor: VolumeActor,
+  voiRange: [number, number]
+) => {
+  const newVoiRange = volumeActor
+    .getProperty()
+    .getRGBTransferFunction(0)
+    .getRange();
+
+  if (newVoiRange[0] === voiRange[0] && newVoiRange[1] === voiRange[1]) {
```

Comment:
I'm confused, don't we just want to fire a voi modified change regardless here? This is just for the initial voi of the viewport, like the first stack new image.

When the volume actor is created, we wanted to emit a voi modified event so that things like overlays can display the initial values. Here if no callback is given, a voi is still set by default, but we aren't emiting a value, so UI etc can't update.

I think we don't need a check here, and we don't need to calculate the old voi. Just after the (optional) callback, find whatever the current voi is, and fire the voi_modified event.